### PR TITLE
Adds Blaze dashboard standalone app

### PR DESCRIPTION
**Pull request by sbarbosa**
Related to SELFSERVE-412

## Proposed Changes

We are making the Advertising dashboard compatible to be used from a Standalone app, that will be consumed inside Jetpack (just like the stats page).

This PR adds a first version of the standalone app (based on the work done in the stats page), and also applies changes to make the advertising page compatible with Jetpack.

### Technical details of the changes:
**DSP Fetches to the API**
I changed to use the library `calypso/lib/wp` instead of directly calling the `request` method of the `wpcom-proxy-request`. It is the same on the inside because the wp library uses the `wpcom-proxy-request` handler to make the calls.

**Dynamic Advertising section URL prefix**
We need a dynamic URL prefix for the Advertising section because we won't use that prefix inside Jetpack (and on Woo in the future). I added a utility method that generates the final URL by getting the URL prefix from a configuration property.
I only spread the changes inside our Advertising module; there are links outside this module that still uses the `/advertising` hardcode prefix, but those aren't a problem because they won't line inside Jetpack or Woo.

**Refactor of the controller.js in promote-post-i2 section**
Because of how Webpack works in the Blaze Dashboard app, we need to be careful with the imports in the promote post section. 
I moved the `redirectToPrimarySite` function from the `controller.js` to the `index.js` because one of the components used in that function forced Webpack to bundle components that we don't need, and some of them conflict with the app breaking styles.
For now, I only detected that importing `calypso/controller` will make our final bundle bigger, forcing us to import the Layout component that will conflict with our custom Jetpack Layout. Let's try having all the logic that needs the `calypso/controller` living inside the index.js instead of the controller.

**Blaze Dashboard new App**
I am adding a new application inside the `/apps` folder. This app is based on the `odyssey-stats` app (an app used inside Jetpack to render the stats page from Calypso). 
have this in mind while reviewing the app, it has some changes from the other, and is still not finished.

## Testing Instructions
* Open the Live preview
* Login as A8C user
* Check out Tools > Advertising
* Verify that the page loads the products/campaigns correctly
* Click on the Promote button on one of the posts
* Verify that the DSP create campaign widget loads correctly
* Click on Cancel and then confirm the modal
* Verify that you get redirected to the post page.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: apps/blaze-dashboard/README.md
Comment (timur987) [inline apps/blaze-dashboard/README.md]: Probably, `intecept` should be replaced with `intercept`, and `covert` with `convert`.

File: apps/blaze-dashboard/bin/build-languages.js
Comment (timur987) [inline apps/blaze-dashboard/bin/build-languages.js]: Just a sanity check, do we really want to use Odyssey's strings .pot file?

File: apps/blaze-dashboard/package.json
Comment (timur987) [inline apps/blaze-dashboard/package.json]: And here as well, do we want to use Odyssey's .pot file?

**Review by therocket-gr — APPROVED** [review]
tested and everything works
code changes LGTM

File: apps/blaze-dashboard/src/load-config.js
Comment (timur987) [inline apps/blaze-dashboard/src/load-config.js]: Probably we need to put the path related to another source of the configData (currently, it's for admin-stats).

File: client/my-sites/promote-post-i2/controller.js
Comment (timur987) [inline client/my-sites/promote-post-i2/controller.js]: Do we need this comment?

**Review by timur987 — APPROVED** [review]
Thanks, @sbarbosa!

I checked changes (they look great) and how they work in Calypso Live and locally.

I am not confident how it should work via jetpack (it seems testing instructions do not cover this topic, but at least I'm sure that the current behavior works as expected).

File: apps/blaze-dashboard/bin/build-languages.js
Comment (sbarbosa) [reply apps/blaze-dashboard/bin/build-languages.js]: You are right, I will rename the file here. Thanks!

File: apps/blaze-dashboard/src/load-config.js
Comment (sbarbosa) [reply apps/blaze-dashboard/src/load-config.js]: I was waiting to merge the PR related to Jetpack, and then go and update all the references in the comments here.
I don't know if we can point to a draft PR here (before it is merged).

I can add TODO in those comments and remove the links until the code is merged.
What do you think @timur987 ?

Comment (timur987) [reply apps/blaze-dashboard/src/load-config.js]: > I can add TODO in those comments and remove the links until the code is merged.

It sounds like a plan. That's definitely a good idea.

File: client/my-sites/promote-post-i2/controller.js
Comment (sbarbosa) [reply client/my-sites/promote-post-i2/controller.js]: Thanks for pointing this out!.
For consistency, I will use the same logic that we use in the Post item component. In there, the source is extracted from the URL query parameter `source`

File: apps/blaze-dashboard/src/load-config.js
Comment (benzhu56) [inline apps/blaze-dashboard/src/load-config.js:14]: Are we forcing to use v2 of campaign dashboard and v2 of blaze widget here? I don't know much about standalone apps in Calypso. Is the current version a standalone app? We are also trying to test the v2 widget redesign with Jetpack, but I'm concerned that with forcing to use `/promote-v2/widget.js` here, we'll expose v2 to the users before releasing it.

Comment (sbarbosa) [reply apps/blaze-dashboard/src/load-config.js:14]: We are forcing both of them here because right now, both are using the same feature flag, and by the time we merge this PR there was an error that if you used the `promote/widget.js` entry point, with the `isV2` flag in true it showed you an mix version of the widget (new redesign HTML but with old CSS). 

This app is not being used anywhere yet, we plan to release the first version when we finish all the redesign tasks and enable it for users in Jetpack after all the testing is done. 

I just tested it and the bug I mentioned before appears to be fixed, so we can make a PR removing this line, which will force the app to take the config value from production instead (that is the v1 entry point).

**Pull request merged**
